container/lxd/lxd.go

```diff
@@ -123,7 +123,9 @@ func (manager *containerManager) CreateContainer(
 		return nil, nil, errors.Trace(err)
 	}
 
-	userData, err := containerinit.CloudInitUserData(instanceConfig, networkConfig)
+	// Do not pass networkConfig, as we want to directly inject our own ENI
+	// rather than using cloud-init.
+	userData, err := containerinit.CloudInitUserData(instanceConfig, nil)
 	if err != nil {
 		return
```

Comment:
I stand to be corrected :-) but think it's OK. err is already declared in the signature and err here is not in a new lexical block, so is only assigned.

I am not keen to introduce a mixture of bare and explicit returns in this function, if we don't need to, but I'll let the second reviewer you ask for push me that way too :-)